packages/loopback/test/acceptance/routing/routing.acceptance.ts

```diff
@@ -90,6 +94,68 @@ describe('Routing', () => {
     expect(result).to.have.property('body', 'TestApp');
   });
 
+  it('creates a new child context for each request', async () => {
+    const app = givenAnApplication();
+    app.bind('flag').to('original');
+
+    const spec = givenOpenApiSpec()
+      .withOperationReturningString('put', '/flag', 'setFlag')
+      .withOperationReturningString('get', '/flag', 'getFlag')
+      .build();
+
+    @api(spec)
+    class FlagController {
+      constructor(@inject('ctx') private ctx: Context) {
```

Comment:
This is the test injecting the full context.

Here is the problem I am trying to solve: I need to write a test that fails when the same app context is used for all requests. To detect that, I need  one of the following:

 1. Some way how to modify a binding in the (per-request) context - that's what I am doing now

 2. Use advanced async/timing control ensure that two requests run in parallel are executed in lockstep. Then I can inject the request object to the controller, write a method that returns the request url including the query string, and use a different query string in each request value. When the timing is right, the controller method executed for the first request will receive the second request from `@inject`, and the test fails. This is not possible at the moment, because I have no control over the async flow inside Router/Context/invoke. I think this could be achieved if context supported async bindings.

Can you think of any better way how to test per-request context instances?